src/test/pages/EventDetails.test.tsx

```diff
@@ -0,0 +1,229 @@
+    await setupTest({ role: "STUDENT", event: mockEvent });
+    const { googleCalendarButton } = getFields();
+    expect(googleCalendarButton()).not.toBeInTheDocument();
+  });
+
+  it("shows event date and time, and message if event was in the past", async () => {
+    const startDate = new Date(new Date().setMonth(new Date().getMonth() - 2));
+    const endDate = new Date(startDate.getTime() + 2 * 60 * 60 * 1000);
+    const pastEvent = { ...mockEvent, date: startDate, endDate: endDate };
+    await setupTest({
+      role: "STUDENT",
+      event: pastEvent,
+    });
+    const { eventDate } = getFields();
+    const eventDateText = `${FRIENDLY_DATE_AND_TIME.format(startDate)} — ${TIME_ONLY.format(endDate)}`;
+    expect(eventDate()).toHaveTextContent(eventDateText);
+    expect(eventDate()).toHaveTextContent("This event is in the past.");
+  });
+
+  it("if event is in the future, no past event warning shows", async () => {
+    const startDate = new Date(new Date().setMonth(new Date().getMonth() + 2));
+    const endDate = new Date(startDate.getTime() + 2 * 60 * 60 * 1000);
+    const futureEvent = { ...mockEvent, date: startDate, endDate: endDate };
+    await setupTest({
+      role: "STUDENT",
+      event: futureEvent,
+    });
+    const { eventDate } = getFields();
+    expect(eventDate()).not.toHaveTextContent("This event is in the past.");
+  });
+
+  it("if event has available places, number of available spaces is shown, and `book a place` button is shown for logged in users", async () => {
+    const event = { ...mockEvent, placesAvailable: 10 };
+    await setupTest({ role: "STUDENT", event });
+    const { placesAvailable } = getFields();
+    expect(placesAvailable()).toHaveTextContent("10 spaces");
+    expect(getButton("Book a place")).toBeInTheDocument();
+  });
+
+  it("if event has no available places, Full badge and `join waiting list` button is shown for logged in users", async () => {
+    const event = { ...mockEvent, placesAvailable: 0 };
+    await setupTest({ role: "STUDENT", event });
+    const { placesAvailable } = getFields();
+    expect(placesAvailable()).toHaveTextContent("Full");
+    expect(getButton("Join waiting list")).toBeInTheDocument();
+  });
+
+  it("if event has no available places and is WAITING_LIST_ONLY, `request a place` button is shown for logged in users", async () => {
+    const event = { ...mockEvent, placesAvailable: 0, eventStatus: "WAITING_LIST_ONLY" as EventStatus };
+    await setupTest({ role: "STUDENT", event });
+    const { placesAvailable } = getFields();
+    expect(placesAvailable()).toHaveTextContent("Full");
+    expect(getButton("Request a place")).toBeInTheDocument();
+  });
+
+  it("if event has no available places and user is RESERVED, `complete reservation` and `cancel registration` buttons are shown for logged in users", async () => {
+    const event = {
+      ...mockEvent,
+      placesAvailable: 0,
+      eventStatus: "WAITING_LIST_ONLY" as EventStatus,
```

Comment:
What you've done is not wrong, but this can also be changed to `as const`. You don't need to change it, but I wanted to mention it because it may be good to spend some time looking into const assertions in TypeScript. I know I have to, as I have never really taken the time to fully understand them.

Reply:
I know I could also use "as const" however there shouldn't be any possibility of the eventStatus changing during the test, so I think it makes more sense to assign it to the correct type that we would expect it to be, unless I've misunderstood the usage!